diag_manager/fms_diag_file_object.F90

```diff
@@ -0,0 +1,407 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+!> @defgroup fms_diag_output_yaml_mod fms_diag_output_yaml_mod
+!> @ingroup diag_manager
+!! @brief fms_diag_file_object_mod handles the file objects data, functions, and subroutines.
+!! @author Tom Robinson
+!! @description The fmsDiagFile_type contains the information for each history file to be written.  It has
+!! a pointer to the information from the diag yaml, additional metadata that comes from the model, and a
+!! list of the variables and their variable IDs that are in the file.
+module fms_diag_file_object_mod
+!use mpp_mod, only: mpp_error, FATAL
+use fms2_io_mod, only: FmsNetcdfFile_t, FmsNetcdfUnstructuredDomainFile_t, FmsNetcdfDomainFile_t
+#ifdef use_yaml
+use fms_diag_yaml_mod, only: diag_yaml, diagYamlObject_type
+
+#endif
+
+implicit none
+private
+
+public :: fmsDiagFile_type, FMS_diag_files
+
+integer, parameter :: var_string_len = 25
+
+type :: fmsDiagFile_type
+ private
+  integer :: id !< The number associated with this file in the larger array of files
+  class(FmsNetcdfFile_t), allocatable :: fileobj !< fms2_io file object for this history file 
+  character(len=1) :: file_domain_type !< (I don't think we will need this)
+#ifdef use_yaml
+  type(diagYamlObject_type), pointer :: diag_yaml => null() !< Pointer to the diag_yaml data
```

Comment:
This should point to `diagYamlFiles_type`. 

The `diagYamlObject_type` has the info of all the files and all the diag_fields.

Reply:
@uramirez8707 How do you suggest I point at the file information only?

Reply:
can you not just do?

```F90
obj%diag_file => diag_yaml%diag_files(i)
```

Reply:
Yes.  When I initially coded it that way, this didn't work.  But now that I've made a bunch of changes all over the place, it worked.